gpkit/constraints/relax.py

```diff
@@ -0,0 +1,142 @@
+                               [[posynomials <= self.relaxvars],
+                                self.relaxvars >= 1],
+                               substitutions)
+
+
+class ConstantsRelaxed(ConstraintSet):
+    """Relax constraints optimally, as in Eqn. 11 of [Boyd2007].
+
+    Arguments
+    ---------
+    constraints : iterable
+        Constraints which will be relaxed (made easier).
+
+    include_only : set
+        if declared, variable names must be on this list to be relaxed
+
+    exclude : set
+        if declared, variable names on this list will never be relaxed
+
+
+    Attributes
+    ----------
+    relaxvars : Variable
+        The variables controlling the relaxation. A solved value of 1 means no
+        relaxation was necessary or optimal for a particular constant.
+        Higher values indicate the amount by which that constant has been
+        made easier: e.g., a value of 1.5 means it was made 50 percent easier
+        in the final solution than in the original problem. Of course, this
+        can also be determined by looking at the constant's new value directly.
+    """
+    def __init__(self, constraints, include_only=None, exclude=None):
+        exclude = frozenset(exclude) if exclude else frozenset()
+        include_only = frozenset(include_only) if include_only else frozenset()
+        if not isinstance(constraints, ConstraintSet):
+            constraints = ConstraintSet(constraints)
+        substitutions = KeyDict(constraints.substitutions)
+        constants, _, _ = parse_subs(constraints.varkeys,
+                                     constraints.substitutions)
+        relaxvars, relaxation_constraints = [], []
+        self.origvars = []
+        self.num = MODELNUM_LOOKUP["Relax"]
+        for key, value in constants.items():
+            if include_only and key.name not in include_only:
+                continue
+            if key.name in exclude:
+                continue
+            descr = dict(key.descr)
+            descr.pop("value", None)
+            descr["units"] = "-"
+            descr["models"] = descr.pop("models", [])+["Relax"]
+            descr["modelnums"] = descr.pop("modelnums", []) + [self.num]
+            relaxation = Variable(**descr)
+            relaxvars.append(relaxation)
+            del substitutions[key]
+            original = Variable(**key.descr)
+            self.origvars.append(original)
+            if original.units and not hasattr(value, "units"):
+                value *= original.units
+            value = Monomial(value)  # convert for use in constraint
+            relaxation_constraints.append([value/relaxation <= original,
+                                           original <= value*relaxation,
+                                           relaxation >= 1])
+        self.relaxvars = NomialArray(relaxvars)
+        ConstraintSet.__init__(self, [constraints, relaxation_constraints])
+        self.substitutions = substitutions
```

Comment:
we might also want a `ConstantsRelaxedEqually`? and/or is there an easy way to make it from the existing classes?